airflow/lineage/__init__.py

```diff
@@ -133,13 +134,16 @@ def wrapper(self, context, *args, **kwargs):
 
             # Remove auto and task_ids
             self.inlets = [i for i in self.inlets if not isinstance(i, str)]
-            _inlets = self.xcom_pull(context, task_ids=task_ids, dag_id=self.dag_id, key=PIPELINE_OUTLETS)
+            with create_session() as session:
+                _inlets = self.xcom_pull(
+                    context, task_ids=task_ids, dag_id=self.dag_id, key=PIPELINE_OUTLETS, session=session
+                )
```

Comment:
Would be a good idea to add a comment explaining why this additional session creation is needed, and why it eliminates the issue (the dangling session in `LazyXComAccess`).

Reply:
Done, I hope the explanation is clear?

Reply:
I update the explanation a bit. As to why the session is not properly closed, I actually do not know. Looking at the code again I can not figure out a reason why it would not close. In the code itself of `_get_bound_query` in the `LazyXComAccess` there is a try finally that closes the session.

Reply:
This is good enough I think. We don’t really know why the leak happens, but know this session does leak sometimes, for watever reason. Programming is hard.